### PR TITLE
Thread Page Fix For Frontend and Backend

### PR DESCRIPTION
**Pull request by Harrylever**

### CODE REVIEW

File: frontend/zc_messaging/src/pages/threads/threads.utils.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ const getThreadHandler = async (orgId, senderId) => {
   try {
     if (orgId && senderId) {
       const getThreadRequest = await axios.get(
+        // `${BASE_URL}/org/${orgId}/member/${senderId}/threads`
```
Comment (JulianaSau) [inline frontend/zc_messaging/src/pages/threads/threads.utils.js:15]: Remove this commented out code

File: frontend/zc_messaging/src/component/ThreadList/index.jsx [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-import styles from "./threadList.module.css"
+import { Container } from "./threadList.style"
 
 const ThreadList = ({ threadListData }) => (
-  <div className={styles.threadListWrap}>
-    {threadListData.map(thread => (
-      <ThreadItem key={thread.message_id} Thread={thread} />
+  <Container>
+    {threadListData.map((message, index) => (
+      <ThreadItem key={index} Thread={message} />
```
Comment (JulianaSau) [inline frontend/zc_messaging/src/component/ThreadList/index.jsx:8]: Use a better key. It's not advisable to use the index from the array directly like that. Make it more unique to the thread items

File: frontend/zc_messaging/src/component/ThreadList/ThreadItem/index.jsx [hunk 3]
```diff
@@ -1,25 +1,36 @@
...
         <p>{richUiData.blocks[0].text}</p>
       </div>
-      <div className={styles.emoji_cont_54x4}>
+      <div className="emoji_cont_54x4">
         <p>{emojis}</p>
       </div>
-    </div>
+      {threads.map((thread, index) => (
+        <div
+          key={index.toString()}
```
Comment (JulianaSau) [inline frontend/zc_messaging/src/component/ThreadList/ThreadItem/index.jsx:24]: Same thing here for this index

**Review by JulianaSau — CHANGES_REQUESTED** [review]

**Comment by JulianaSau** [comment]
@mykie88 Please help to review the backend code in this PR

File: frontend/zc_messaging/src/component/ThreadList/ThreadItem/index.jsx (hunk 3 above)
Comment (Harrylever) [reply frontend/zc_messaging/src/component/ThreadList/ThreadItem/index.jsx:24]: Honestly, there's no key of the thread object that has a unique attribute value to fit in as the key. But I'll see if I still change it.

File: frontend/zc_messaging/src/component/ThreadList/index.jsx (hunk 2 above)
Comment (Harrylever) [reply frontend/zc_messaging/src/component/ThreadList/index.jsx:8]: Okay

File: backend/endpoints/threads.py [hunk 4]
```diff
@@ -235,7 +235,7 @@ async def get_threads(org_id: str, member_id: str, page: int = 1, size: int = 15
         room_ids.append(room['_id'])
 
     for room_id in room_ids:
-        single_room_resp = await get_room_messages(org_id, room_id, page, size)
+        single_room_resp = await get_room_messages_for_threads(org_id, room_id)
```
Comment (mikengr) [inline backend/endpoints/threads.py:238]: `get_room_threaded_messages` is a better name for this function and the variable should `threaded_messages`

Also I don't think this your for loop is doing the correct logic as intended

File: backend/endpoints/threads.py [hunk 5]
```diff
@@ -235,7 +235,7 @@ async def get_threads(org_id: str, member_id: str, page: int = 1, size: int = 15
...
         if single_room_resp == []:
             # Room not found
             pass
```
Comment (mikengr) [inline backend/endpoints/threads.py:241]: this if statement should be joined with the next one

File: frontend/zc_messaging/src/component/ThreadList/ThreadItem/index.jsx (hunk 3 above)
Comment (JulianaSau) [reply frontend/zc_messaging/src/component/ThreadList/ThreadItem/index.jsx:24]: You could simply use the thread ID that comes with the messages.

File: backend/utils/message_utils.py [hunk 6]
```diff
@@ -103,6 +103,59 @@ async def get_room_messages(
     return response
 
 
+async def get_room_messages_for_threads(
```
Comment (mikengr) [inline backend/utils/message_utils.py:106]: this function will just return the same result as `get_room_messages` excluding pagination. If you want to write a utility function that returns threaded messages, then the logic should be different from that go get room messages.

File: backend/utils/message_utils.py [hunk 7]
```diff
@@ -81,7 +81,7 @@ async def get_room_messages(
     DB = DataStorage(org_id)
 
     skip = await off_set(page, size)
-    options = {"limit": size, "skip": skip, "sort": { "created_at": -1}}
+    options = {"limit": size, "skip": skip, "sort": { "created_at": 1}}
```
Comment (mikengr) [inline backend/utils/message_utils.py:84]: is there a reason for modifying the sort option?

**Review by mikengr — CHANGES_REQUESTED** [review]
@Harrylever I left some comments
Also, it seems you have been deactivated, what happened?

**Pull request closed without merging**